DataFormats/CSCDigi/src/CSCShowerDigi.cc

```diff
@@ -35,6 +53,42 @@ bool CSCShowerDigi::isNominalOutOfTime() const { return bitsOutOfTime() >= kNomi
 
 bool CSCShowerDigi::isTightOutOfTime() const { return bitsOutOfTime() >= kTight; }
 
+bool CSCShowerDigi::isValidShowerType() const { return showerType_ > kInvalidShower; }
+
 std::ostream& operator<<(std::ostream& o, const CSCShowerDigi& digi) {
-  return o << "CSC Shower: in-time bits " << digi.bitsInTime() << ", out-of-time bits " << digi.bitsOutOfTime();
+  unsigned int showerType = digi.getShowerType();
+  std::string compHitsStr(", comparatorHits ");
+  compHitsStr += std::to_string(digi.getComparatorNHits());
+  std::string wireHitsStr(", wireHits ");
+  wireHitsStr += std::to_string(digi.getWireNHits());
+  std::string showerStr;
+  switch (showerType) {
+    case 0:
+      showerStr = "Invalid ShowerType";
+      break;
+    case 1:
+      showerStr = "AnodeShower";
+      break;
+    case 2:
+      showerStr = "CathodeShower";
+      break;
+    case 3:
+      showerStr = "MatchedShower";
+      break;
+    case 4:
+      showerStr = "EMTFShower";
+      break;
+    case 5:
+      showerStr = "GMTShower";
+      break;
+    default:
+      showerStr = "UnknownShowerType";
+  }
+
+  //return o << (showerType == 0
+  //                 ? "Invalid ShowerType"
+  //                 : (showerType == 1 ? "ALCTShower" : (showerType == 2 ? "CLCTShower" : "correlatedLCTShower")))
```

Comment:
commented-out code should be removed

Reply:
@jpata Done in [5eb6603](https://github.com/cms-sw/cmssw/pull/38941/commits/5eb66037a3fe46e47876ae64512ddcb10a5cdc49)